include/flatbuffers/flatbuffers.h

```diff
@@ -1248,6 +1262,28 @@ class FlatBufferBuilder
   }
   #endif
 
+  /// @brief Serialize an array of structs into a FlatBuffer `vector`.
+  /// @tparam T The data type of the struct array elements.
+  /// @param[in] f A function that takes the current iteration 0..vector_size-1,
+  /// a pointer to the struct that must be filled and the state argument.
+  /// @param[in] state Arbitrary state to pass to f.
+  /// @return Returns a typed `Offset` into the serialized data indicating
+  /// where the vector is stored.
+  /// This is mostly useful when flatbuffers are generated with mutation
+  /// accessors.
+  template <typename T, typename F, typename S>
+  Offset<Vector<const T *>> CreateVectorOfStructs(size_t vector_size, F f,
+                                                  S *state) {
+    StartVector(vector_size * sizeof(T) / AlignOf<T>(), AlignOf<T>());
```

Comment:
Apart from creating a pair of CreateVectorOfStructStart() / CreateVectorOfStructEnd() methods to handle the couple of lines of pre / post code around the loop.  I don't think so, that of course adds the potential for the compiler to not inline this code in addition makes it a little more tricky to read.  Any other suggestions?